keep/step/step.py

```diff
@@ -1,31 +1,53 @@
+import asyncio
+from enum import Enum
+import inspect
 import logging
 from dataclasses import field
 
 import chevron
 from pydantic.dataclasses import dataclass
 
+from keep.conditions.condition_factory import ConditionFactory
 from keep.contextmanager.contextmanager import ContextManager
+from keep.exceptions.action_error import ActionError
 from keep.iohandler.iohandler import IOHandler
 from keep.providers.base.base_provider import BaseProvider
+from keep.throttles.throttle_factory import ThrottleFactory
+
+class StepType(Enum):
+    STEP = "step"
+    ACTION = "action"
 
 
 @dataclass(config={"arbitrary_types_allowed": True})
 class Step:
-    step_id: str
-    step_config: dict
-    provider: BaseProvider
-    provider_parameters: dict
+    name: str = field(default_factory=str)
+    config: dict = field(default_factory=dict)
+    step_id: str = field(default_factory=str)
+    step_config: dict = field(default_factory=dict)
+    provider: BaseProvider = field(default_factory=BaseProvider)
+    provider_context: dict = field(default_factory=dict)
+    provider_parameters: dict = field(default_factory=dict)
+    conditions_results: dict = field(default_factory=dict)
+    conditions: list = field(default_factory=list)
+    step_type: Enum = field(default=StepType.STEP)
 
     def __post_init__(self):
         self.io_handler = IOHandler()
         self.logger = logging.getLogger(__name__)
         self.context_manager = ContextManager.get_instance()
+        self.conditions = self.config.get("condition", [])
 
     @property
     def foreach(self):
         return self.step_config.get("foreach")
-
+    
     def run(self):
+        if self.step_type == StepType.STEP:
```

Comment:
if action and steps are the same, why is there a "_run_step" and "_run_action"? what's the difference between them?

Reply:
> if action and steps are the same, why is there a "_run_step" and "_run_action"? what's the difference between them?

Only difference is calling notify or query

Reply:
so it should be something like:
```
class Runnable():
    def run(...):
         return self._run()
    
class Step(Runnable):
     def run(...):
          return self._query()
          
class Action(Runnable):
     def run(...):
          return self._notify()
```

Reply:
and I'm not even sure why we need to distinguish between query and notify, we can consolidate it too

Reply:
@joaquinto and thanks for your contribution! <3

Reply:
@talboren ok. I will look into it.